native/acter/src/api/room.rs

```diff
@@ -545,27 +545,33 @@ impl SpaceRelations {
 
     pub async fn query_hierarchy(&self) -> Result<Vec<SpaceHierarchyRoomInfo>> {
         let c = self.room.core.clone();
-        let room = self.room.room.clone();
+        let suggested_rooms = self
+            .core
+            .children
+            .iter()
+            .filter(|c| c.suggested())
+            .map(|c| c.room_id())
+            .collect::<Vec<_>>();
         let room_id = self.room.room_id().to_owned();
         RUNTIME
             .spawn(async move {
-                let relations = c.space_relations(&room).await?.children.iter().filter(|c| c.suggested()).map(|c| c.room_id()).collect::<Vec<_>>();
                 let mut next : Option<String> = Some("".to_owned());
                 let mut rooms = Vec::new();
                 while next.is_some() {
-                    let request = assign!(get_hierarchy::v1::Request::new(room_id.clone()), { from: next.clone(), max_depth: Some(1u32.into()) });
```

Comment:
okay, this is the last part of the actual fix: we were not really always properly loading all subsequent children, because we limited the `max_depth=1`. As this is a tree-like-structure, the _parent_ we are asking for is the root level of 1. Thus, we need to ask for root-level of `2` to actually get all the proper children information. I am not quite sure how this has worked before to be honest ... it puzzles me a bit...